source/dotnet/Test/AdaptiveCards.Test/SerializationTests.cs

```diff
@@ -256,6 +256,95 @@ public void TestStyleNullDeserialization()
             Assert.IsNotNull(result.Card);
         }
 
+        [TestMethod]
+        public void Test_TypeHandling()
+        {
+            AdaptiveCard card = new AdaptiveCard("1.0")
+            {
+                Body =
+                {
+                    new AdaptiveTextBlock("Hello world"),
+                    new AdaptiveImage("http://adaptivecards.io/content/cats/1.png"),
+                    new AdaptiveColumnSet()
+                    {
+                        Columns = new List<AdaptiveColumn>()
+                        {
+                            new AdaptiveColumn()
+                            {
+                                Width = "32px",
+                                Items = new List<AdaptiveElement>()
+                                {
+                                    new AdaptiveTextBlock("1")
+                                    {
+                                        Wrap = true,
+                                        Size = AdaptiveTextSize.Large,
+                                        HorizontalAlignment = AdaptiveHorizontalAlignment.Center,
+                                        Color = AdaptiveTextColor.Accent
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            };
+
+
+            // make card into JObject with types included
+            JObject cardObject = JObject.FromObject(card, new Newtonsoft.Json.JsonSerializer()
+            {
+                TypeNameHandling = Newtonsoft.Json.TypeNameHandling.All
+            });
+
+            // now bring it back
+            AdaptiveCard card2 = cardObject.ToObject<AdaptiveCard>();
+
+            // card2 will now have AdditionalProperties because $type is not known and it seems $type is not ignored by Newtonsoft JsonExtensionData
```

Comment:
@clintwar should this be considered a bug we should look into?

Reply:
@golddove not sure I would call it a bug, since AdaptiveTypeElement has AdditionalProperties with JsonExtensionData attribute anything not defined gets thrown into it. $type is included by Newtonsoft when you serialize but I do not see any good means to ignore since there is not a good way to differentiate things included by the serialization vs things introduced in the json directly.